### PR TITLE
feat: add stack order modal

### PR DESCRIPTION
**Pull request by Diogomartf**
## Description
This PR adds stack modal to stacks table.

### summary
- add stack modal
  - order details 
  - orders progress bar
- add models folder to handle model related functions and types
  -  add order model 
- add utils folder
  - time utils
  - token utils
- updates Modal component
  - sharing type props
  - better spacing
  - handles width better with max-width
  
  
**Screenshot**
  
image

### CODE REVIEW

File: packages/app/app/models/order/index.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export * from "./order";
```
Comment (berteotti) [inline packages/app/app/models/order/index.ts:1]: Should we move this folder `models` outside the `app`? This looks something that can live in the same level as components

File: packages/app/components/OrdersProgressBar.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { OrderProps, totalOrdersDone, totalOrders } from "@/app/models/order";
+import React, { useRef, useEffect } from "react";
+
+export const OrdersProgressBar = ({ order }: OrderProps) => {
```
Comment (berteotti) [inline packages/app/components/OrdersProgressBar.tsx:4]: Should we do the same with this component? Seems to be a very custom component that can be defined inside `StackProgress`.

File: packages/app/app/models/order/index.ts (hunk 1 above)
Comment (berteotti) [reply packages/app/app/models/order/index.ts:1]: Also this is shared between pages and components. So makes senso to live in the same level.

File: packages/app/components/OrdersProgressBar.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  const progressBarRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (progressBarRef.current) {
+      const totalWidth = progressBarRef.current.offsetWidth;
+      const width = (totalWidth * totalOrdersDone(order)) / totalOrders(order);
+      progressBarRef.current.style.width = `${width}px`;
+    }
+  }, [order]);
```
Comment (berteotti) [inline packages/app/components/OrdersProgressBar.tsx:13]: Isn't calculating a percentage with better than this? No need for refs, and we can use tailwind to define the custom percentage width.

File: packages/app/components/stack-modal/StackProgress.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    <div className="flex flex-col justify-between space-y-1 md:space-y-0 md:items-center md:flex-row">
+      <OrdersExecuted order={order} />
+      <div className="flex items-center space-x-1">
+        <BodyText size="responsive" className="text-em-low">
+          Total funds used:{" "}
+          <span className="text-em-high">
+            {totalFundsUsed(order).toFixed(2)}{" "}
+            <span className="text-xs text-em-low">of</span>{" "}
+            {fundsUsedWithToken(order)}
+          </span>
```
Comment (berteotti) [inline packages/app/components/stack-modal/StackProgress.tsx:21]: This looks a bit different from the design, any reason in particular?

File: packages/app/components/stack-modal/StackProgress.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import { OrdersProgressBar } from "@/components/OrdersProgressBar";
+import { BodyText } from "@/ui";
+
+export const StackProgress = ({ order }: OrderProps) => (
+  <div className="mt-3 space-y-3">
+    <div className="flex flex-col justify-between space-y-1 md:space-y-0 md:items-center md:flex-row">
+      <OrdersExecuted order={order} />
+      <div className="flex items-center space-x-1">
+        <BodyText size="responsive" className="text-em-low">
+          Total funds used:{" "}
```
Comment (berteotti) [inline packages/app/components/stack-modal/StackProgress.tsx:16]:
```suggestion
          Total Funds used:{" "}
```

I like how the design looks :)

File: packages/app/components/stack-modal/StackProgress.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        <div className="w-4 h-4 rounded-full bg-primary-100"></div>
+      </div>
+    </div>
+    <OrdersProgressBar order={order} />
+  </div>
+);
+
+const OrdersExecuted = ({ order }: OrderProps) => {
+  if (!totalOrdersDone(order))
+    return <BodyText className="text-em-low">No orders executed yet</BodyText>;
```
Comment (berteotti) [inline packages/app/components/stack-modal/StackProgress.tsx:32]:
```suggestion
    return <BodyText className="text-em-low">No orders executed yet.</BodyText>;
```

Just a subtle change.

File: packages/app/utils/time.ts [hunk 7]
```diff
@@ -1,8 +1,13 @@
-export const currentTimestampInSeconds = Math.floor(Date.now() / 1000);
```
Comment (berteotti) [inline packages/app/utils/time.ts:1]: Suggestion for this file name: `datetime.ts` or `date-time.js`

**Review by berteotti — CHANGES_REQUESTED** [review]
Looks good, just left some comments :)

File: packages/app/components/stack-modal/StackProgress.tsx (hunk 6 above)
Comment (Diogomartf) [reply packages/app/components/stack-modal/StackProgress.tsx:32]: Fixed. thks

File: packages/app/utils/time.ts (hunk 7 above)
Comment (Diogomartf) [reply packages/app/utils/time.ts:1]: 👍

Comment (Diogomartf) [reply packages/app/utils/time.ts:1]: datetime.js (I think in this case makes sense to be together)

File: packages/app/components/stack-modal/StackProgress.tsx (hunk 4 above)
Comment (Diogomartf) [reply packages/app/components/stack-modal/StackProgress.tsx:21]: was that the difference you noticed?

File: packages/app/components/OrdersProgressBar.tsx (hunk 2 above)
Comment (Diogomartf) [reply packages/app/components/OrdersProgressBar.tsx:4]: Actually this one will be used on other parts, that's why I add it to components. For eg. when we do the custom OrderCards on mobile.

image

File: packages/app/components/stack-modal/StackProgress.tsx (hunk 5 above)
Comment (Diogomartf) [reply packages/app/components/stack-modal/StackProgress.tsx:16]: I don't think that capitalizing funds make sense. I prefer: Total funds used:

File: packages/app/app/models/order/index.ts (hunk 1 above)
Comment (Diogomartf) [reply packages/app/app/models/order/index.ts]: fixed: https://github.com/SwaprHQ/stackly-ui/pull/23/commits/d950260c05cfac42e9a6372d7572331783ca866b

File: packages/app/components/OrdersProgressBar.tsx (hunk 3 above)
Comment (Diogomartf) [reply packages/app/components/OrdersProgressBar.tsx:13]: I tried at runtime but couldn't make it work with tailwind custom properties. Maybe I'm missing something. 

But from a quick search:

> Tailwind doesn’t include any sort of client-side runtime, so class names need to be statically extractable at build-time, and can’t depend on any sort of arbitrary dynamic values that change on the client

File: packages/app/components/stack-modal/StackProgress.tsx (hunk 5 above)
Comment (berteotti) [reply packages/app/components/stack-modal/StackProgress.tsx:16]: I think it's a reading hack, where you put more emphasis on what matters (which is Total Funds), but it works in both ways for me.

File: packages/app/components/stack-modal/StackProgress.tsx (hunk 4 above)
Comment (Diogomartf) [reply packages/app/components/stack-modal/StackProgress.tsx]: yeah, this part is getting big and the "of" is not important, so I made it smaller and decreased the emphasis.

image

File: packages/app/components/stack-modal/StackProgress.tsx (hunk 5 above)
Comment (Diogomartf) [reply packages/app/components/stack-modal/StackProgress.tsx:16]: I bite this. Updated.

**Review by berteotti — APPROVED** [review]

**Pull request merged**